mlx/primitives.cpp

```diff
@@ -2174,6 +2181,21 @@ std::vector<array> Scatter::vjp(
               gather(cotangents[0], indices, axes_, slice_sizes, stream()));
           break;
         }
+        case Scatter::Max: {
+          auto slice_sizes = cotangents[0].shape();
+          for (auto ax : axes_) {
+            slice_sizes[ax] = 1;
+          }
+          auto result = scatter_max(values, indices, updates, axes_, stream());
+          auto gathered_cotan =
+              gather(cotangents[0], indices, axes_, slice_sizes, stream());
+          auto gathered_result =
+              gather(result, indices, axes_, slice_sizes, stream());
+          vjps.push_back(multiply(
+              gathered_cotan,
+              where(gathered_result == updates, array({1}), array({0}))));
```

Comment:
Could you change that to 

```cpp
multiply(gathered_cotan, gathered_result == updates, stream())
```

no need for the `where` and adding the stream.

Reply:
Thanks for the tip, modif is pushed ✅